robottelo/performance/pulp.py

```diff
@@ -0,0 +1,140 @@
+        """retrieve time from stderr"""
+        # should return only one time point as a single sync
+        real_time = ''
+        for line in stderr.split('\n'):
+            if line.startswith('real'):
+                real_time = line
+        return 0 if real_time == '' else float(real_time.split(' ')[1])
+
+    @classmethod
+    def _get_enabled_repos(cls, org_id):
+        """Get all enabled repositories ids and names
+
+        :return map_repo_name_id: The dictionary contains all enabled
+            repositories in Satellite. Map repo-name as key, repo-id as value
+
+        """
+        LOGGER.info('Get All enabled repositories from database:')
+
+        result = Repository.list(
+            {'organization-id': org_id},
+            per_page=False
+        )
+
+        if result.return_code != 0:
+            LOGGER.info('Fail to fetch enabled repos.')
+            raise RuntimeError('No enabled repository found!')
+        # map repository name with id
+        map_repo_name_id = {}
+        for repo in result.stdout:
+            map_repo_name_id[repo['name']] = repo['id']
+        return map_repo_name_id
+
+    @classmethod
+    def repositories_sequential_sync(
+            cls,
+            repo_names_list,
+            map_repo_name_id,
+            sync_iterations,
+            savepoint=None):
+        """Sync all repositories linearly, and repeat X times
+
+        :param list repo_names_list: A list of targeting repository names
+        :param int sync_iterations: The number of times to repeat sync
+        :return time_result_dict_sync
+        :rtype: dict
+
+        """
+        # Create a dictionary to store all timing results from each sync
+        time_result_dict_sync = {}
+
+        # repeat sequential sync X times
+        for i in range(sync_iterations):
+            # note: name key by thread to adapt to graph module
+            key = 'thread-{}'.format(i)
+            time_result_dict_sync[key] = []
+
+            # Sync each repo one-by-one and collect timing data
+            for repo_name in repo_names_list:
+                repo_id = map_repo_name_id.get(repo_name, None)
+                if repo_id is None:
+                    LOGGER.warning('Invalid repository name!')
+                    continue
+                LOGGER.debug(
+                    'Sequential Sync {0} attempt {1}:'.format(repo_name, i)
+                )
+                # sync repository once at a time
+                time_result_dict_sync[key].append(
+                    cls.repository_single_sync(repo_id, repo_name, 'linear')
+                )
+            # for resync purpose, no need to restore
+            if savepoint is None:
+                return
+            else:
+                # restore database at the end of each iteration
+                cls._restore_from_savepoint(savepoint)
+
+        return time_result_dict_sync
+
+    @classmethod
+    def _restore_from_savepoint(cls, savepoint):
```

Comment:
This method uses nothing on the class. Why bother making it a class method? How about making this a static method instead?